lms/static/scripts/frontend_apps/utils/test/onedrive-picker-client-test.js

```diff
@@ -0,0 +1,145 @@
+import { PickerCanceledError } from '../google-picker-client';
+import { OneDrivePickerClient, $imports } from '../onedrive-picker-client';
+
+describe('OneDrivePickerClient', () => {
+  let fakeLoadOneDriveAPI;
+  const clientOptions = {
+    clientId: '12345',
+    redirectURI: 'https://redirect.uri',
+  };
+
+  function createClient({ clientId, redirectURI } = clientOptions) {
+    return new OneDrivePickerClient({ clientId, redirectURI });
+  }
+
+  beforeEach(() => {
+    fakeLoadOneDriveAPI = sinon.stub();
+
+    $imports.$mock({
+      './onedrive-api-client': {
+        loadOneDriveAPI: fakeLoadOneDriveAPI,
+      },
+    });
+    delete window.OneDrive;
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+    delete window.OneDrive;
+  });
+  describe('#constructor', () => {
+    it('calls `loadOneDriveAPI`', () => {
+      createClient();
+
+      // If `loadOneDriveAPI` fails to load the OneDrive client it raises an
+      // unhandled rejection, which possible to emulate and catch on a test but
+      // difficult hide from the output.
+      assert.calledOnce(fakeLoadOneDriveAPI);
+    });
+  });
+
+  describe('#showPicker', () => {
+    it('fails to invoke `window.OneDrive.open` if OneDrive client failed to load', async () => {
+      const oneDrive = createClient();
+      let expectedError;
+
+      // Emulate failure to load the OneDrive client:
+      // 1. an unhandled rejection raised in the constructor
+      // 2. window.OneDrive is undefined
```

Comment:
The connection between `loadOneDriveAPI` failing and `window.OneDrive` not being defined is implicit. This hazard would be avoided if `showPicker` used the result of `loadOneDriveAPI`. Additionally it would allow `showPicker` to return a more obvious error (eg. "Failed to load OneDrive" or whatever `loadOneDriveAPI` rejects with).